CONTRIBUTING.md

```diff
@@ -0,0 +1,76 @@
+# CONTRIBUTING
+
+## TLDR;
+
+1. Create your own fork of the repo
+2. Make changes to the code in your fork
+3. Submit PR from your fork to main branch of the project repo
+
+## Setting up your development environment
+
+The development should prefer [Python 3.11](https://docs.python.org/3/whatsnew/3.11.html) or later due to significant improvement on performance, optimizations which benefit modern ML, AI, LLM, NL stacks, and improved asynchronous proccessing capabilities.
+
+```bash
+# Clone your fork
+git clone https://github.com/williamcaban/lightspeed-service.git
+
+# move into the directory
+cd lightspeed-service
+
+# Setup your Python Virtual Environment 
+# to avoid conflicts with your system packages
+python3.11 -m venv venv
+
+# Activate the virtual environment
+source ./venv/bin/activate
+
+# Upgrade pip to the most recent version
+pip install --upgrade pip
+
+# Install dependencies
+pip install -r requirements.txt
+```
+
+Happy hacking!
+
+## Before a PR
+
+If updating `requirements.txt` identify if doing so for the `main` branch or a release branch and follow the corresponding guidance. If a dependency is no longer required, remove it from the list.
```

Comment:
i don't understand this part.  If you're updating requirements.txt for main branch, the PR will be against the main branch.

if you're doing it for a release branch (note: we're probably about to delete our release branches for now as they don't make sense right now) you'd submit the PR against the release branch.

so you don't need to "identity" which you're doing it for

Reply:
The intent is to have the guidance for whoever cut the releases.

Reply:
ok then maybe put this in a separate section about cutting releases